src/ru/animelib/src/eu/kanade/tachiyomi/animeextension/ru/animelib/Animelib.kt

```diff
@@ -0,0 +1,500 @@
+        val url = apiUrl.toHttpUrl().newBuilder()
+        url.addPathSegment("constants")
+        url.addQueryParameter("fields[]", "videoServers")
+
+        val videoServerResponse = client.newCall(GET(url.build())).execute()
+        val videoServers = videoServerResponse.parseAs<VideoServerData>()
+
+        val serverPreference = preferences.getString(PREF_SERVER_KEY, PREF_SERVER_ENTRIES[0])
+        if (serverPreference.isNullOrEmpty()) {
+            return videoServers.data.videoServers[0].url
+        }
+
+        for (videoServer in videoServers.data.videoServers) {
+            if (videoServer.label == serverPreference) {
+                return videoServer.url
+            }
+        }
+
+        return videoServers.data.videoServers[0].url
+    }
+
+    private fun kodikVideoLinks(playerUrl: String?, teamName: String): List<Video> {
+        val useKodik = preferences.getBoolean(PREF_USE_KODIK_KEY, PREF_USE_KODIK_DEFAULT)
+        if (playerUrl.isNullOrEmpty() || !useKodik) {
+            return emptyList()
+        }
+
+        val kodikPage = "https:$playerUrl"
+        val headers = Headers.Builder()
+        headers.add("Referer", baseUrl)
+        val kodikPageResponse = client.newCall(GET(kodikPage, headers.build())).execute()
+
+        // Parse form parameters for video link request
+        val page = kodikPageResponse.asJsoup()
+        val urlParams = page.getElementsByTag("script").find {
+            it.html().contains(domain)
+        }?.html() ?: return emptyList()
+
+        val formData = urlParams.substringAfter("urlParams = '")
+            .substringBefore("'")
+            .parseAs<KodikForm>()
+
+        if (formData.dSign.isEmpty()) {
+            return emptyList()
+        }
+
+        val kodikDomain = formData.pd
+        val formBody = FormBody.Builder()
+        formBody.add("d", formData.d)
+        formBody.add("d_sign", URLDecoder.decode(formData.dSign, "utf-8"))
+        formBody.add("pd", formData.pd)
+        formBody.add("pd_sign", URLDecoder.decode(formData.pdSign, "utf-8"))
+        formBody.add("ref", URLDecoder.decode(formData.ref, "utf-8"))
+        formBody.add("ref_sign", URLDecoder.decode(formData.refSign, "utf-8"))
+
+        val urlParts = playerUrl.split('/')
+        formBody.add("type", urlParts[3])
+        formBody.add("id", urlParts[4])
+        formBody.add("hash", urlParts[5])
+
+        val videoInfoRequest = POST("https://$kodikDomain/ftor", body = formBody.build())
+        val videoInfoResponse = client.newCall(videoInfoRequest).execute()
+        val kodikData = videoInfoResponse.parseAs<KodikData>()
+
+        // Load js with encode algorithm and parse it
+        val scriptElement = page.getElementsByTag("script").find {
+            it.outerHtml().contains("player_single")
+        } ?: return emptyList()
+
+        val scriptHref = scriptElement.attr("src")
+        val jsScript = client.newCall(GET("https://$kodikDomain$scriptHref")).execute().body.string()
```

Comment:
```suggestion
        val scriptUrl = page.selectFirst("script[src*=player_single]")?.attr("abs:src")
            ?: return emptyList()
        val jsScript = client.newCall(GET(scriptUrl)).execute().body.string()
```
You can prefix `abs:` on an attr to get the absolute url.